### DISCUSSION TITLE
Unlinking workspace package after linking

### DISCUSSION THREAD

**Discussion by alexandercerutti** [pnpm/pnpm#8874] [Q&A] [answered]
Hello, I have the following situation: in order to run tests like I'm using a workspace package from outside (e.g. to test `exports` in `package.json`), I run these commands:

```json
{
	"scripts": {
     		"test": "pnpm build && pnpm test:prepare && node --test && pnpm test:cleanup",
		"test:prepare": "pnpm link .",
		"test:cleanup": "pnpm unlink"
	},
	"dependencies": {
		"another-package-from-workspace": "workspace:^"
	}
}
```

So after building, I want to link the current package to its `node_modules`, then run the tests (they run correctly) and then I want to cleanup the linking in order to future avoid problems.

However, `pnpm unlink` runs successfully without cleaning anything.

When I run `pnpm unlink .` (with any other argument, like the package name, `pnpm unlink package-name`), I receive the following error:

```sh
ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "another-package-from-workspace@workspace:^" is in the dependencies but no package named "another-package-from-workspace" is present in the workspace
```

But this is not correct because when I install dependencies it works perfectly, tests run perfectly and I can see the dependency installed (linked).

The issue happens both when I run

```sh
# from workspace root
$ pnpm -r test

# from package folder
$ pnpm test
```

I use this way to test local imports on other workspaces (and this doesn't work as well).
However, when I run these commands on pnpm projects that are not workspaces (no monorepos), these commands seems to work fine. It removes the local dependency and it runs correctly.

In that case, the unlink command is this one:

```sh
$ pnpm unlink package-name
```

And if I run

```sh
$ pnpm unlink
```

It doesn't remove anything.

Do you know if there's a way to solve this? Is this a bug?

I already tried with `--filter` as well, but it didn't work, as it seems to be working only from workspace root.

Thank you!

**Comment by zkochan** [comment, marked as answer]
Just add the package as a dependency of itself to devDependencies and don't use link/unlink:

```json
{
  "name": "my-pkg",
  "devDependencies": {
    "my-pkg": "workspace:*"
  }
}
```

We use this technic in the pnpm monorepo.

**Reply by alexandercerutti** [reply]
@zkochan thank you for your response. This works perfectly!
Going to link this to the issue I've opened.
